datastore/google/cloud/datastore/connection.py

```diff
@@ -220,6 +237,44 @@ def allocate_ids(self, project, request_pb):
                          _datastore_pb2.AllocateIdsResponse)
 
 
+def _grpc_catch_rendezvous(to_call, *args, **kwargs):
```

Comment:
I don't like that the invocation of the `to_call` is handed off to `_grpc_catch_rendezvous`, I would much prefer:

```python
with _grpc_catch_rendezvous():
    to_call(...)
```